Base/splitter.h

```diff
@@ -0,0 +1,42 @@
+#ifndef SPLITTER_H__
+#define SPLITTER_H__
+
```

Comment:
I would move this file to something like MachineLearning/tree/splitter. Moreover, we will be using namespace to better separate things. Take a look at the style guide.